TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -364,6 +383,32 @@ public void onClick(View view) {
         }
     }
 
+    public void onClickOkLoadNavMapDialog(final String mapName) {
+        if (mapName == null || mapName.isEmpty()) {
+            Log.e(TAG, "Map name is null or empty, unable to load the navigation map");
+            displayToastMessage(R.string.map_name_error);
+            return;
+        }
+        new AsyncTask<Void, Void, Void>() {
+            @Override
+            protected Void doInBackground(Void... params) {
+                mTangoServiceClientNode.callLoadNavMapService(mapName);
+                return null;
+            }
+        }.execute();
```

Comment:
Make sure that the user cannot load another map while the service is already running. Disable the load map button in this case or open a progress dialog to keep the user waiting.